catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/GeoCoderEndpoint.java

```diff
@@ -58,27 +48,37 @@ public Response getLocation(@QueryParam("jsonp") String jsonp,
         resourceSets.add(resourceSet);
         JSONArray resources = new JSONArray();
         resourceSet.put("resources", resources);
-        JSONObject resource = new JSONObject();
-        JSONArray bbox = new JSONArray();
-        List<Point> points = geoResult.getBbox();
-        DirectPosition upperCorner = points.get(0).getDirectPosition();
-        DirectPosition lowerCorner = points.get(1).getDirectPosition();
-        bbox.add(upperCorner.getCoordinate()[1]);
-        bbox.add(upperCorner.getCoordinate()[0]);
-        bbox.add(lowerCorner.getCoordinate()[1]);
-        bbox.add(lowerCorner.getCoordinate()[0]);
-        resource.put("bbox", bbox);
-        JSONObject point = new JSONObject();
-        point.put("type", "Point");
-        JSONArray coordinates = new JSONArray();
-        coordinates.add(latitude);
-        coordinates.add(longitude);
-        point.put("coordinates", coordinates);
-        resource.put("point", point);
-        resource.put("name", geoResult.getFullName());
-        resources.add(resource);
 
-        response = Response.ok(jsonp + "(" + jsonObject.toJSONString() + ")").build();
-        return response;
+        GeoResult geoResult = geoCoder.getLocation(query);
+
+        if (geoResult != null) {
+            DirectPosition directPosition = geoResult.getPoint().getDirectPosition();
+            double[] coords = directPosition.getCoordinate();
+
+            double longitude = coords[0];
+            double latitude = coords[1];
+
+            JSONObject resource = new JSONObject();
+            JSONArray bbox = new JSONArray();
+            List<Point> points = geoResult.getBbox();
+            DirectPosition upperCorner = points.get(0).getDirectPosition();
+            DirectPosition lowerCorner = points.get(1).getDirectPosition();
+            bbox.add(upperCorner.getCoordinate()[1]);
+            bbox.add(upperCorner.getCoordinate()[0]);
+            bbox.add(lowerCorner.getCoordinate()[1]);
+            bbox.add(lowerCorner.getCoordinate()[0]);
+            resource.put("bbox", bbox);
+            JSONObject point = new JSONObject();
+            point.put("type", "Point");
+            JSONArray coordinates = new JSONArray();
+            coordinates.add(latitude);
+            coordinates.add(longitude);
+            point.put("coordinates", coordinates);
+            resource.put("point", point);
+            resource.put("name", geoResult.getFullName());
+            resources.add(resource);
+        }
+
+        return Response.ok(jsonp + "(" + jsonObject.toJSONString() + ")").build();
```

Comment:
The JavaScript that handles this request is responsible for attaching the appropriate message depending on the response.